include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8876,4 +8876,8 @@ def err_bounds_type_annotation_lost_checking : Error<
   def err_initializer_expected_for_ptr : Error<
     "automatic variable %0 with _Ptr type must have initializer">;
 
+  def err_not_non_modifying_expr : Error<
+	"%select{assignment|increment|decrement|call|comma|volatile}0 expression not allowed "
+	"within non-modifying expression in dynamic_check">;
```

Comment:
This should use the official name, which is "_Dynamic check".      We should not use the term "non-modifying expression" in an error message without some sort of clarification because it is not a well-known term.